sars2seq/translate.py

```diff
@@ -71,3 +140,77 @@ def translate(seq, name=None):
     seq += 'N' * (3 - remainder if remainder else 0)
 
     return Seq(seq).translate()
+
+
+def translateSpike(seq):
+    """
+    Translate a Spike sequence, taking into account gaps introduced in the
+    nucleotide alignment. This means that the amino acid sequences do not
+    have to be re-aligned after translating, which avoids the introduction of
+    gaps at the amino acid level that may be different from gaps at the
+    nucleotide level.
+
+    @param seq: A C{str} nucelotide sequence.
+    @return: A translated C{str} amino acid sequence that retains the gaps.
+    """
+    sequence = ''
+    current = 0
+    while current + 3 <= len(seq):
```

Comment:
Although `len` happens to be fast in Python, there's still no reason to call it (i guess) thousands of times:

```python
seqLen = len(seq)
while current + 3 <= seqLen:
```

Reply:
Done.